Modules/arraymodule.c

```diff
@@ -2852,15 +2875,16 @@ array_iter(arrayobject *ao)
 static PyObject *
 arrayiter_next(arrayiterobject *it)
 {
+    array_state *state = find_array_state_by_type(Py_TYPE(it));
```

Comment:
Since the state is only used in asserts here, could you wrap this in `#ifdef NDEBUG`? (And possibly the asserts themselves as well, to avoid warnings.)

Reply:
Of course! I'll push it when I get to my laptop later today. Thanks!

Reply:
Is f8649699eac6673497be01c4a986f2fc1d144f20 acceptable?

Reply:
Better than I expected! Thank you!

Reply:
Good to hear! Thanks for reviewing!